SprawdzanieNIPu/ConsoleApplication1/NIP.cs

```diff
@@ -1,72 +1,68 @@
-﻿using System;
+﻿namespace ConsoleApplication1
+ {
+    using System;
 
-namespace ConsoleApplication1
-{
-    class NIP
-    {
-        public string numer;
-
-        public NIP(string nr)
-        {
-            this.numer = nr.Trim();
-        }
-
-        public Boolean dlugoscOK
-        {
-            get
-            {
-                return this.numer.Length == 10;
-            }
-        }
-
-        public int kontrolna
-        {
-            // zwraca sumę kontrolną
-            get
-            {
-                int[] wagi = new int[] {6, 5, 7, 2, 3, 4, 5, 6, 7};
-                double x=0;
-                double liczba=0;
+ /// <summary>
+ /// Klasa służy do sprawdzenia, czy wprowadzony ciąg znaków jest poprawnym numerem NIP i jego sformatowania.
+ /// </summary>
+ internal class NIP
+ {
+   private long numer;
+   private int[] wagi = new int[] { 6, 5, 7, 2, 3, 4, 5, 6, 7 };
+   private int kontrolna;
 
-                for (int i = 0; i < wagi.Length; i++) 
-                {
-                    x = Convert.ToDouble(numer.Substring(i, 1));
-                    liczba = liczba + x*wagi[i];
-                }
+   private bool czyPoprawny;
+   private string sformatowana;
 
-                liczba = liczba % 11;
+   public bool CzyPoprawny
```

Comment:
Generalnie przyjąłbym takie podejście, że jak będziesz miał 3 lata doświadczenia jako developer .NET, to będziesz mógł wyłączać w StyleCopie co zechcesz :)